iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/RunTreeImplTest.java

```diff
@@ -0,0 +1,132 @@
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.common.api.tree.TextTree;
+import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
+import org.sonar.iac.docker.parser.utils.Assertions;
+import org.sonar.iac.docker.tree.api.RunTree;
+import org.sonar.iac.docker.tree.api.DockerTree;
+import org.sonar.iac.docker.tree.api.ExecFormLiteralTree;
+import org.sonar.iac.docker.tree.api.ExecFormTree;
+import org.sonar.iac.docker.tree.api.LiteralListTree;
+import org.sonar.iac.docker.tree.api.SeparatedList;
+import org.sonar.iac.docker.tree.api.ShellFormTree;
+import org.sonar.iac.docker.tree.api.SyntaxToken;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class RunTreeImplTest {
+
+  @Test
+  void shouldParseRunExecForm() {
+    Assertions.assertThat(DockerLexicalGrammar.RUN)
+      .matches("RUN")
+      .matches("RUN []")
+      .matches("RUN [\"ls\"]")
+      .matches("RUN [\"executable\",\"param1\",\"param2\"]")
+      .matches("RUN [\"/usr/bin/wc\",\"--help\"]")
+      .matches("RUN [\"/usr/bin/wc\",\"--help\"]")
+      .matches("    RUN []")
+      .matches("RUN [\"c:\\\\Program Files\\\\foo.exe\"]")
+      .matches("run")
+
+      .notMatches("RUND")
+      // not exec form
+      .notMatches("");
+  }
+
+  @Test
+  void shouldParseRunShellForm() {
+    Assertions.assertThat(DockerLexicalGrammar.RUN)
+      .matches("RUN")
+      .matches("RUN ls")
+      .matches("RUN \"ls\"")
+      .matches("RUN command param1 param2")
+      .matches("RUN echo \"This is a test.\" | wc -")
+      .matches("RUN /bin/sh /deploy.sh")
+      .matches("RUN mkdir -p /output && zip -FS -r /output/lambda.zip ./")
+      .matches("RUN \"/usr/bin/run.sh\"")
+      .matches("    RUN \"/usr/bin/run.sh\"")
+      .matches("RUN     \"/usr/bin/run.sh\"")
+      .matches("run")
+      // not exec form
+      .matches("RUN [\"la\", \"-bb\"")
+      .matches("RUN [\"la\", \"-bb]")
+      .matches("RUN \"la\", \"-bb\"]")
+
+      .notMatches("/bin/sh /deploy.sh");
+  }
+
+  @Test
+  void shouldCheckParseRunExecFormTree() {
+    RunTree tree = DockerTestUtils.parse("RUN [\"executable\",\"param1\",\"param2\"]", DockerLexicalGrammar.RUN);
+    assertThat(tree.getKind()).isEqualTo(DockerTree.Kind.RUN);
+    assertThat(tree.keyword().value()).isEqualTo("RUN");
+
+    assertThat(tree.arguments()).isNotNull();
+    assertThat(tree.arguments().type()).isEqualTo(LiteralListTree.LiteralListType.EXEC);
+    assertThat(tree.arguments().literals().stream().map(TextTree::value)).containsExactly("\"executable\"", "\"param1\"", "\"param2\"");
+    assertThat(((SyntaxToken)tree.children().get(0)).value()).isEqualTo("RUN");
+
+    assertThat(tree.children().get(1)).isInstanceOf(ExecFormTree.class);
```

Comment:
Please add at least one assertTextRange, to show that all children are taken into account when calculating the tree text range

Reply:
Sure I will improve this